HLTrigger/JetMET/plugins/HLTScoutingPFProducerV2.cc

```diff
@@ -0,0 +1,293 @@
+      pfJetTagCollection_(consumes(iConfig.getParameter<edm::InputTag>("pfJetTagCollection"))),
+      pfCandidateCollection_(consumes(iConfig.getParameter<edm::InputTag>("pfCandidateCollection"))),
+      vertexCollection_(consumes(iConfig.getParameter<edm::InputTag>("vertexCollection"))),
+      metCollection_(consumes(iConfig.getParameter<edm::InputTag>("metCollection"))),
+      rho_(consumes(iConfig.getParameter<edm::InputTag>("rho"))),
+      pfJetPtCut_(iConfig.getParameter<double>("pfJetPtCut")),
+      pfJetEtaCut_(iConfig.getParameter<double>("pfJetEtaCut")),
+      pfCandidatePtCut_(iConfig.getParameter<double>("pfCandidatePtCut")),
+      pfCandidateEtaCut_(iConfig.getParameter<double>("pfCandidateEtaCut")),
+      mantissaPrecision_(iConfig.getParameter<int>("mantissaPrecision")),
+      doJetTags_(iConfig.getParameter<bool>("doJetTags")),
+      doCandidates_(iConfig.getParameter<bool>("doCandidates")),
+      doMet_(iConfig.getParameter<bool>("doMet")),
+      doCandIndsForJets_(iConfig.getParameter<bool>("doCandIndsForJets")) {
+  //register products
+  produces<Run3ScoutingPFJetCollection>();
+  produces<Run3ScoutingParticleV2Collection>();
+  produces<double>("rho");
+  produces<double>("pfMetPt");
+  produces<double>("pfMetPhi");
+}
+
+HLTScoutingPFProducerV2::~HLTScoutingPFProducerV2() = default;
+
+// ------------ method called to produce the data  ------------
+void HLTScoutingPFProducerV2::produce(edm::StreamID sid, edm::Event &iEvent, edm::EventSetup const &setup) const {
+  using namespace edm;
+
+  //get vertices
+  Handle<reco::VertexCollection> vertexCollection;
+  std::unique_ptr<Run3ScoutingVertexCollection> outVertices(new Run3ScoutingVertexCollection());
+  if (iEvent.getByToken(vertexCollection_, vertexCollection)) {
+    for (auto const& vtx : *vertexCollection) {
+      outVertices->emplace_back(MiniFloatConverter::reduceMantissaToNbitsRounding(vtx.x(), mantissaPrecision_),
+                                MiniFloatConverter::reduceMantissaToNbitsRounding(vtx.y(), mantissaPrecision_),
+                                MiniFloatConverter::reduceMantissaToNbitsRounding(vtx.z(), mantissaPrecision_),
+                                MiniFloatConverter::reduceMantissaToNbitsRounding(vtx.zError(), mantissaPrecision_),
+                                MiniFloatConverter::reduceMantissaToNbitsRounding(vtx.xError(), mantissaPrecision_),
+                                MiniFloatConverter::reduceMantissaToNbitsRounding(vtx.yError(), mantissaPrecision_),
+                                vtx.tracksSize(),
+                                MiniFloatConverter::reduceMantissaToNbitsRounding(vtx.chi2(), mantissaPrecision_),
+                                vtx.ndof(),
+                                vtx.isValid());
+    }
+  }
+
+  //get rho
+  Handle<double> rho;
+  std::unique_ptr<double> outRho(new double(-999));
+  if (iEvent.getByToken(rho_, rho)) {
+    outRho = std::make_unique<double>(*rho);
```

Comment:
```suggestion
    *outRho = *rho;
```
would be (tiny bit) cheaper.